db/write_batch_internal.h

```diff
@@ -260,4 +260,173 @@ class LocalSavePoint {
 #endif
 };
 
+template <typename Checker>
+class TimestampAssignerBase : public WriteBatch::Handler {
+ public:
+  explicit TimestampAssignerBase(WriteBatch::ProtectionInfo* prot_info,
+                                 Checker&& checker)
+      : prot_info_(prot_info), checker_(std::move(checker)) {}
+
+  ~TimestampAssignerBase() override {}
+
+  Status PutCF(uint32_t cf, const Slice& key, const Slice&) override {
+    Status s = AssignTimestamp(cf, key);
+    ++idx_;
```

Comment:
I think `idx_` would be in a better place in `SimpleListTimestampAssigner` (and could be incremented in `SimpleListTimestampAssigner::AssignTimestamp`)

Reply:
It's also needed for `prot_info_` which is needed by `TimestampAssigner`.

Reply:
Thanks, I missed that